pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -613,6 +614,65 @@ func (s *discovery) updateServicesCPUStats(services []model.Service) error {
 	return nil
 }
 
+func (s *discovery) enrichContainerData(service *model.Service, containers []*agentPayload.Container, pidToCid map[int]string) {
+	id, ok := pidToCid[service.PID]
+	if !ok {
+		return
+	}
+
+	// The tags we look for service name generation, in their priority order.
+	// The map entries will be filled as we go through the containers tags.
+	tagsPriority := []struct {
+		tagName  string
+		tagValue *string
+	}{
+		{"service", nil},
+		{"app", nil},
+		{"short_image", nil},
+		{"kube_container_name", nil},
+		{"kube_deployment", nil},
+		{"kube_service", nil},
+	}
+
+	service.ContainerID = id
+	log.Debugf("Found container id for %v: %v", service.Name, id)
+	for _, c := range containers {
+		if c.Id != id {
```

Comment:
Do you think it would be worth it to create a temporary id-to-container map to avoid looping through the container list for every service?